src/obfuscation/hidden.c

```diff
@@ -0,0 +1,123 @@
+#include "hidden.h"
+#include "rmalloc.h"
+#include "util/minmax.h"
+#include "redis_index.h"
+#include "query_node.h"
+#include "reply_macros.h"
+
+typedef struct {
+  const char *user;
+  uint64_t length;
+} UserString;
+
+HiddenName *NewHiddenName(const char* name, uint64_t length, bool takeOwnership) {
+  UserString* value = rm_malloc(sizeof(*value));
+  value->user = name;
+  if (takeOwnership) {
+    value->user = rm_strndup(name, length);
+  }
+  value->length = length;
+  return (HiddenName*)value;
+};
+
+void HiddenName_Free(HiddenName* hn, bool tookOwnership) {
+  UserString* value = (UserString*)hn;
+  if (tookOwnership) {
+    rm_free((void*)value->user);
+  }
+  rm_free(value);
+};
+
+static inline int Compare(const char *left, size_t left_length, const char *right, size_t right_length) {
+  int result = strncmp(left, right, MIN(left_length, right_length));
+  if (result != 0 || left_length == right_length) {
+    return result;
+  } else {
+    return left_length < right_length ? -1 : 1;
```

Comment:
Consider subtracting

Reply:
Worried about casting size_t to a signed type.